Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,356 @@
+    ArgumentUtility.CheckNotNullAndTypeIsObject('searchStringValidationInfo', searchStringValidationInfo);
+    ArgumentUtility.CheckNotNullAndTypeIsString('nullValueString', nullValueString);
+    ArgumentUtility.CheckNotNullAndTypeIsBoolean('isAutoPostBackEnabled', isAutoPostBackEnabled);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('searchContext', searchContext);
+    ArgumentUtility.CheckNotNullAndTypeIsBoolean('isIconUpdateEnabled', isIconUpdateEnabled);
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsObject('iconContext', iconContext);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('resources', resources);
+
+    this._itemBackUp = null;
+    this._isInvalidated = false;
+    this.BackupItemData(hiddenField.val(), textbox.val());
+    this._command = command;
+    this._commandBackUp = command;
+    this._selectListID = baseID + '_Results';
+    this._informationPopUpID = baseID + '_Information';
+    this._isAutoPostBackEnabled = isAutoPostBackEnabled;
+    this._nullValueString = nullValueString;
+    this._isIconUpdateEnabled = isIconUpdateEnabled;
+    this._controlServiceUrl = controlServiceUrl;
+    this._iconContext = iconContext;
+    this._commandInfo = commandInfo;
+
+    if (BrowserUtility.GetIEVersion() > 0)
+    {
+      // For Internet Explorer + JAWS, we must use the ARIA 1.0 combobox pattern.
+      // The remaining browsers support ARIA 1.1 with NVDA.
+
+      var internetExplorerScreenReaderLabelID = baseID + '_InternetExplorerScreenReaderLabel';
+      var screenReaderLabel = $("<span hidden='hidden' />")
+        .attr("id", internetExplorerScreenReaderLabelID)
+        .html(resources.InternetExplorerScreenReaderLabelText)
+        .appendTo(combobox);
+
+      textbox.attr('role', combobox.attr('role'));
+      combobox.removeAttr('role');
+
+      textbox.attr('aria-haspopup', combobox.attr('aria-haspopup'));
+      combobox.removeAttr('aria-haspopup');
+
+      textbox.attr('aria-expanded', combobox.attr('aria-expanded'));
+      combobox.removeAttr('aria-expanded');
+
+      if (combobox.attr('aria-labelledby') !== undefined)
+      {
+        textbox.attr('aria-labelledby', combobox.attr('aria-labelledby') + ' ' + internetExplorerScreenReaderLabelID);
+        combobox.removeAttr('aria-labelledby');
+
+        if (combobox.attr('data-label-id-index') !== undefined)
+        {
+          textbox.attr('data-label-id-index', combobox.attr('data-label-id-index'));
+          combobox.removeAttr('data-label-id-index');
+        }
+      }
+      else
+      {
+        textbox.attr('aria-labelledby', internetExplorerScreenReaderLabelID);
+      }
+
+      if (combobox.attr('aria-describedby') !== undefined)
+      {
+        textbox.attr('aria-describedby', combobox.attr('aria-describedby'));
+        combobox.removeAttr('aria-describedby');
+      }
+
+      combobox = textbox;
+    }
+
+    const ClearError = () =>
```

Comment:
If `ClearError` becomes an instance method, `textbox` must become an instance variable and `clearRequestError` passed to `this._textbox.autocomplete` must become a lambda, otherwise it looks like it can't capture the `this` context, but if we do that, it is possible

Reply:
Yes, the instance-field for `textbox` would beneeded. `clearRequestError` I don't yet understand. quick session before/after standup?